include/spdlog/details/os-inl.h

```diff
@@ -458,8 +462,12 @@ SPDLOG_INLINE void wstr_to_utf8buf(wstring_view_t wstr, memory_buf_t &target)
         return;
     }
 
+#ifdef SPDLOG_USE_STD_FORMAT
+    int result_size = 0;
```

Comment:
`.data()` on std::basic_string is only valid for the range `[data(), data() + size())` so using the capacity here would make WideCharToMultiByte write outside of this range. This will be visible once the string is resized, because everything past the previous size is going to be set to `\0`, erasing what WideCharToMultiByte just wrote.

Reply:
> so using the capacity here would make WideCharToMultiByte write outside of this range

not sure how this can happen. the string object resized in line 477 to the right size prior to WideCharToMultiByte (the first WideCharToMultiByte doesn’t write anything to the target and only used to get the expected size).

Reply:
You are right. Will fix next commit.